fhir2dataset/fhirpath.py

```diff
@@ -0,0 +1,112 @@
+"""set of functions allowing to use the javascript coded library on the repository https://github.com/HL7/fhirpath.js
+"""  # noqa
+import json
+import logging
+from subprocess import Popen, PIPE
+
+from fhir2dataset.timer import timing
+
+logger = logging.getLogger(__name__)
+
+wrapper = """
+(function run(globals) {
+    try {
+        var result = (%(func)s).apply(globals, %(args)s);
+        if ((typeof result) == 'string') {
+            result = JSON.stringify(result);
+        }
+        console.log(JSON.stringify({"result": result}))
```

Comment:
I think it would be better to return the string instead of using console.log.

Reply:
Currently, I launch from python a pipe on which I run the js script and I think I can only read the script's response if it's written in the pipe (which the console.log does). If I make a return in the current state of the code, I can't retrieve the result. :confused:

Reply:
hmm okay I see, it's not really elegant.. But I don't know if there is another way to do that

You should comment to explain that in your code :)

Reply:
@nriss Why don't you think it's elegant? Maybe by explaining to me why I could find a method that would suit you better :)
Basically the function I use works like a terminal on our computer and therefore can only send and read messages exchanged on this kind of terminal. Would there be another method to write to the terminal from which you run a javascript script?
In any case, it's a good idea to add the explanation in the code smile

Reply:
According to me, console.log is just a way to print data but not to share it.

Moreover, you cannot totally control what will be written on the pipe : for example, imagine that in some case, the fhirpath.js library also print some console.log

Reply:
I made a change to prevent this type of behavior :smile: